### PR TITLE
domain: load all non-public tables into infoschema

### LINKED ISSUE #54906: full load doesn't load non-public table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

loadTableInfo will load non-public tables which fetchSchemasWithTables doesn't. Which is inconsistent.

### 2. What did you expect to see? (Required)
N/A
### 3. What did you see instead (Required)
N/A
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by joechenrh**
### What problem does this PR solve?

Issue Number: close #54906

### What changed and how does it work?
Currently, `fetchSchemasWithTables` will load non-public schema. This can cause inconsistence between v1 and v2 after `drop database` and V1V2 consistency check has found several failed cases related to this.

Consider the following sequence of operations:

```
drop database --> Change to WriteOnly --> Full Load --> Change to DeleteOnly --> ApplyDiff
```

`applyDropSchemaV2` will return all tableIDs in this deleted schema while `applyDropSchema` will return nothing.

https://github.com/pingcap/tidb/blob/ebbe53c98b57e71239cd794c8fd9bb4861f35c98/pkg/infoschema/infoschema_v2.go#L1246-L1264

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

You can test it using unit test in this PR: #55030

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: pkg/infoschema/infoschema_v2.go [hunk 1]
```diff
@@ -1043,6 +1043,14 @@ func loadTableInfo(ctx context.Context, r autoid.Requirement, infoData *Data, tb
 			))
 		}
 
+		// table is not public
+		if tblInfo.State != model.StatePublic {
```
Comment (tiancaiamao) [inline pkg/infoschema/infoschema_v2.go:1047]: Does it means something wrong if we see the non-public tables?

Do you have any case to cover it? or this is just 'defensive programming'

Comment (joechenrh) [reply pkg/infoschema/infoschema_v2.go:1047]: Maybe the only case is: user would see table being deleted from `information_schema.tables`.

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**